common/source/docs/common-dshot.rst

```diff
@@ -40,17 +43,29 @@ The DShot protocol can run at different speeds. ArduPilot supports four speeds:
+For larger aircraft with longer cable runs we recommend using the lowest baud rate, DShot150, as it is the most reliable protocol (lower baudrates are less susceptible to noise on cables). Higher rates can be more susceptible to noise but also tie up the allocated DMA channel for a shorter period so can be beneficial on flight controllers with a lot of DMA sharing.
+For BLHeli32 ESCs - used mainly on smaller craft - DShot600 is by far the most widely used and can therefore be a more suitable choice simply because of the amount of testing that it has had.
+
+Bi-directional DShot involves a longer pulse width since it has to wait for a response from the ESC before it can send another pulse and thus DShot300 and DShot600 are to be preferred. Bi-directional DShot does not share DMA channels and so there is no impact on other peripherals.
 
 .. note:: When an output is configured for DShot, the ``SERVOx_MIN/MAX/TRIM`` parameters for that output will always be ignored since DShot does not use these parameters. The trim  value used will be  1500 if it's a reversible output, or 1000 if normal output setup in DShot, and the output range always be 1000-2000.
 
+The frequency at which DShot pulses are sent can be configured through :ref:`SERVO_DSHOT_RATE<SERVO_DSHOT_RATE>`. By default ArduPilot will output a DShot pulse every time a new IMU sample is ready *and* at a fixed 1Khz interval. On a copter with the standard 400Hz scheduler loop rate this works out at about 1.4Khz. However, the output is quite irregular - in order to get more regular output :ref:`SERVO_DSHOT_RATE<SERVO_DSHOT_RATE>` can be configured to send pulses at multiples of the scheduler loop rate. Thus if set to 2 the pulses will be sent at 800Hz, set to 3 at 1.2Khz and so on. The difference being very, very even output which can benefit copters needing tighter motor control (for instance smaller racers). It is not recommended to send pulses at less than 1Khz due to reports of BLHeli32 occasionally missing frames on some flight controllers, similarly sending at higher rates can result in increased reliability and faster recovery from missed pulses where needed at the cost of some CPU. Very high rates can only be used for faster DShot speeds since otherwise pulses might overlap - for instance the fastest rate that DShot150 can theoretically support is 4Khz.
+
+DShot Commands
+--------------
+
+On certain ESCs DShot commands are supported. These allow functions such as ESC LEDs, beeps and motor direction to be manipulated by the flight controller. In order to use DShot commands :ref:`SERVO_DSHOT_ESC<SERVO_DSHOT_ESC>` must be set to the type of ESC that is in use. Notify functions (e.g. LEDs and Buzzer) can then be configure to include DShot as an output type.
```

Comment:
what happens if you set this and the ESC does not support the commands or not all escs support them? nothing?

Reply:
The command set for other ESCs is different so undefined - so its important to set the type correctly

Reply:
but right now there is only one type of command set....if set to 1 for an ESC wihtout Dshot command capability, what happens? and if set to 1 for an ESC that does not have the command set ("BLHeli32/BLHeli_S/Kiss" ) what happens? smoke? explosions? nothing?

Reply:
It depends on the ESC - on some nothing, on others maybe a beep instead of an LED flash. This only came up because the ESCs that tridge is working with have a slightly different command set. I suppose the worse that could happen is that 3D is selected - or not selected - unintentionally.

Reply:
sorry to keep on this, but are there any features , besides notify, that require the SERVO_DSHOT_ESC be set to 1? you imply reversible(3D)....or can this be set using passthru?if so it should be noted also...
perhaps something like this should be added: " Currently, ArduPilot supports the command set (SERVO_DSHOT_ESC=1) that is commonly used, however, others are appearing and may not be compatible, resulting in undefined operation. Use caution (remove blades!) until correct operation using type=1 is verified for your ESC."

Reply:
Seems reasonable. Currently supported features:
- Dshot LEDs via NTF_LED_TYPES
- Dshot Buzzer via NTZ_BUZZ_TYPES
- Reverse motor direction via RVMASK
- Enable 3D operation via 3DMASK

Both of the latter can be set via passthru but obviously requires a different tool be fired up so more convenient this way